src/react/components/pages/projectSettings/projectForm.tsx

```diff
@@ -125,10 +103,61 @@ export default class ProjectForm extends React.Component<IProjectFormProps, IPro
                         type="button"
                         onClick={this.onFormCancel}>{strings.common.cancel}</button>
                 </div>
+                <TagEditorModal
+                    ref={this.tagEditorModal}
+                    onOk={this.onTagModalOk}
+
+                    tagNameText={strings.tags.modal.name}
+                    tagColorText={strings.tags.modal.color}
+                    saveText={strings.common.save}
+                    cancelText={strings.common.cancel}
+                />
             </Form>
         );
     }
 
+    private fields() {
+        return {
+            sourceConnection: CustomField(ConnectionPicker, (props) => {
+                return {
+                    id: props.idSchema.$id,
+                    value: props.formData,
+                    connections: this.props.connections,
+                    onChange: props.onChange,
+                };
+            }),
+            targetConnection: CustomField(ConnectionPicker, (props) => {
+                const targetConnections = this.props.connections.filter(
+                    (connection) => StorageProviderFactory.isRegistered(connection.providerType));
+                return {
+                    id: props.idSchema.$id,
+                    value: props.formData,
+                    connections: targetConnections,
+                    onChange: props.onChange,
+                };
+            }),
+            tagsInput: CustomField(TagsInput, (props) => {
+                const tagsInputProps: ITagsInputProps = {
+                    tags: props.formData,
+                    onChange: props.onChange,
+                    placeHolder: strings.tags.placeholder,
+                    onCtrlTagClick: this.onTagClick,
+                    ref: this.tagsInput,
```

Comment:
why do we need this `ref` here?

Reply:
@wbreza there was a reason for that, and I remember discussing this together, but it's been a little bit, so I'm a little fuzzy on the reason. I think the `ref` needs to be in the properties of the `Widget` that's created for the `Form` component. Before it wasn't necessary because we weren't creating a reference in the form, but with the change to decouple the `TagsInput` and the `TagEditorModal`, we needed to create a ref for each.

Reply:
I actually don't think that we want this RefObject & ref props pattern. I would like to refactor it if possible.

Reply:
Will address on another ticket.